packages/openapi-validator/lib/classes/OpenApi3Spec.ts

```diff
@@ -88,7 +88,7 @@ export default class OpenApi3Spec extends AbstractOpenApiSpec {
     return { components: this.getComponentDefinitions() };
   }
 
-  getSchemaObjects(): OpenAPIV3.ComponentsObject['schemas'] {
+  getSchemaObjects(): Pick<OpenAPIV3.ComponentsObject, 'schemas'> {
```

Comment:
Thanks @tom-arnold for raising this and submitting a fix! 

I can't reproduce the error. Does this error come from `jestOpenAPI(spec)` or `toSatisfySchemaInApiSpec(<whatsThisObject>)`? Would you mind pasting what code triggers the error, your full stack trace, tsconfig, and OpenAPI doc version please? 

For fastest resolution, please recreate this in [our recreation template](https://github.com/openapi-library/OpenAPIValidators/blob/master/packages/jest-openapi/__test__/bugRecreationTemplate.test.ts)

I ask because I'm not sure how this fixes your error. I assume you want `'{ [key: string]: ReferenceObject | SchemaObject; } | undefined'` to change to `'{ [key: string]: ReferenceObject | SchemaObject; }'`.

But [given](https://github.com/kogosoftwarellc/open-api/blob/7beed9d37cd367cd6682b6c91b3af9bb801aeb10/packages/openapi-types/index.ts#L520):
```js
interface ComponentsObject {
        schemas?: {
            [key: string]: ReferenceObject | SchemaObject;
        };
```

`OpenAPIV3.ComponentsObject['schemas']` is `{ [key: string]: ReferenceObject | SchemaObject; } | undefined`, 

whereas `Pick<OpenAPIV3.ComponentsObject, 'schemas'>` is 
```
{
        schemas?: {
            [key: string]: ReferenceObject | SchemaObject;
        };
}
```